runtime/queries/metricsview_timeseries.go

```diff
@@ -78,117 +78,22 @@ func (q *MetricsViewTimeSeries) Resolve(ctx context.Context, rt *runtime.Runtime
 		return err
 	}
 
-	switch olap.Dialect() {
-	case drivers.DialectDuckDB:
-		return q.resolveDuckDB(ctx, rt, instanceID, mv, priority)
-	case drivers.DialectDruid:
-		return q.resolveDruid(ctx, olap, mv, priority)
-	default:
-		return fmt.Errorf("not available for dialect '%s'", olap.Dialect())
-	}
-}
-
-func (q *MetricsViewTimeSeries) Export(ctx context.Context, rt *runtime.Runtime, instanceID string, w io.Writer, opts *runtime.ExportOptions) error {
-	err := q.Resolve(ctx, rt, instanceID, opts.Priority)
-	if err != nil {
-		return err
-	}
-
-	mv, err := lookupMetricsView(ctx, rt, instanceID, q.MetricsViewName)
-	if err != nil {
-		return err
-	}
-
-	if opts.PreWriteHook != nil {
-		err = opts.PreWriteHook(q.generateFilename(mv))
-		if err != nil {
-			return err
-		}
-	}
-
-	tmp := make([]*structpb.Struct, 0, len(q.Result.Data))
-	meta := append([]*runtimev1.MetricsViewColumn{{
-		Name: mv.TimeDimension,
-	}}, q.Result.Meta...)
-	for _, dt := range q.Result.Data {
-		dt.Records.Fields[mv.TimeDimension] = structpb.NewStringValue(dt.Ts.AsTime().Format(time.RFC3339Nano))
-		tmp = append(tmp, dt.Records)
-	}
-
-	switch opts.Format {
-	case runtimev1.ExportFormat_EXPORT_FORMAT_UNSPECIFIED:
-		return fmt.Errorf("unspecified format")
-	case runtimev1.ExportFormat_EXPORT_FORMAT_CSV:
-		return writeCSV(meta, tmp, w)
-	case runtimev1.ExportFormat_EXPORT_FORMAT_XLSX:
-		return writeXLSX(meta, tmp, w)
-	case runtimev1.ExportFormat_EXPORT_FORMAT_PARQUET:
-		return writeParquet(meta, tmp, w)
+	tr := &runtimev1.TimeSeriesTimeRange{
+		Start:    q.TimeStart,
+		End:      q.TimeEnd,
+		Interval: q.TimeGranularity,
 	}
 
-	return nil
-}
-
-func (q *MetricsViewTimeSeries) generateFilename(mv *runtimev1.MetricsView) string {
-	filename := strings.ReplaceAll(mv.Model, `"`, `_`)
-	if q.TimeStart != nil || q.TimeEnd != nil || q.Filter != nil && (len(q.Filter.Include) > 0 || len(q.Filter.Exclude) > 0) {
-		filename += "_filtered"
-	}
-	return filename
-}
-
-func (q *MetricsViewTimeSeries) resolveDuckDB(ctx context.Context, rt *runtime.Runtime, instanceID string, mv *runtimev1.MetricsView, priority int) error {
-	ms, err := resolveMeasures(mv, q.InlineMeasures, q.MeasureNames)
+	ntr, err := resolveNormaliseTimeRange(ctx, rt, instanceID, priority, mv.Model, mv.TimeDimension, tr)
```

Comment:
This still maintains coupling (the coupling is problematic for the new granular access policies).

But is time range normalization even necessary for the metrics view API? I think it's okay to require the UI to pass a time granularity.